Updates for GET and POST

311chaos opened the pull request:
- updated to handle the events for _handleAdd and _handleAll, these now get tossed back to the main component to deal with the _images array.
- Passing in the data for addImage using FormData()
- Updated the demo page to allow for mocking some data (there is some weirdness if you have already added images through the API)
- Updated tests

311chaos committed:
Updates for GET and POST

(now on a branch)

* DELETE is failing, need to verify route and any header data.

* updated to handle the events for _handleAdd and _handleAll, these now get tossed back to the main component to deal with the _images array.

* Passing in the data for addImage using FormData()

* Updated the demo page to allow for mocking some data (there is some weirdness if you have already added images through the API)

* Updated tests

311chaos commented:
I will keep this open until I have added in the DELETE and the POST and done a little refactoring.

311chaos committed:
adding manager tests

* defaulting _images array
* tests support GET and POST flows

311chaos committed:
a few more tests

311chaos committed:
Updated Docs

311chaos committed:
Delete works along with a test

311chaos committed:
fixed dialog height issue.

311chaos committed:
refactor

* id to _id
* name to fileName
* updating some tests that were getting events triggered from other tests
* removed extraneous reference to sinon in the test files
* Calling PUT on images-api

311chaos committed:
a few small changes

* fixing an issue with overwriting the tags array.
* support for PUT
* Test

311chaos committed:
support for sort

311chaos committed:
sortOrder, setAsDefault changes

* sortOrder is now updated immediately after dropping.
* setAsDefaultImage is now passing more details through
* updated tests to not fire updateImage network requests when running the drag/drop tests.

311chaos committed:
passing the url as an attribute

zack37 commented:
Uploading image in the demo doesn't work

zack37 commented on ll-property-images.html:
```diff
@@ -25,32 +25,38 @@
           description="{{item.description}}"
           src="{{item.url}}"
-          is-default-image$="{{item.isDefaultImage}}"
+          is-default$="{{item.isDefault}}"
```
The $= maybe be unnecessary since it's for attributes, not properties.

311chaos committed:
removed $ for attribute binding

311chaos committed:
images to images-api

load images is EXPLICIT now, this is to avoid the requests being fired on tests/demos

311chaos committed:
giving an event to hook up to

zack37 commented:
Mostly looks good. I'm still having trouble uploading in the demo, but we can talk about that later. One thing that would be nice to add is documentation for events fired, specifically for not the manager. I see the manager has events documented.

311chaos commented:
It's because the images Api was changed to "images-api" but is not working in prod anymore. (Under images or images-api).

We need to sync up with Colton and Jason to see if that is pending another PR.

Sent from Outlookhttp://aka.ms/Ox5hz3

On Wed, Sep 2, 2015 at 7:11 AM -0700, "Zack Smith" <notifications@github.com<mailto:notifications@github.com>> wrote:

Mostly looks good. I'm still having trouble uploading in the demo, but we can talk about that later. One thing that would be nice to add is documentation for events fired.

## 

Reply to this email directly or view it on GitHubhttps://github.com/LeisureLinkElements/ll-property-images/pull/2#issuecomment-137094919.

311chaos committed:
Updating docs for events

311chaos commented:
@zack37 - Updates made. Can you leave this branch open after the merge?
